pkg/controller/volume/expand/expand_controller.go

```diff
@@ -208,16 +208,21 @@ func (expc *expandController) pvcUpdate(oldObj, newObj interface{}) {
 		}
 
 		// Filter PVCs for which the corresponding volume plugins don't allow expansion.
-		volumeSpec := volume.NewSpecFromPersistentVolume(pv, false)
-		volumePlugin, err := expc.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
-		if err != nil || volumePlugin == nil {
-			err = fmt.Errorf("didn't find a plugin capable of expanding the volume; " +
-				"waiting for an external controller to process this PVC")
-			expc.recorder.Event(newPVC, v1.EventTypeNormal, events.ExternalExpanding,
-				fmt.Sprintf("Ignoring the PVC: %v.", err))
-			glog.V(3).Infof("Ignoring the PVC %q (uid: %q) : %v.",
-				util.GetPersistentVolumeClaimQualifiedName(newPVC), newPVC.UID, err)
-			return
+		if volumeSpec := volume.NewSpecFromPersistentVolume(pv, false); volumeSpec.PersistentVolume.Spec.FlexVolume == nil {
+			// A flex volume is expandable but since it resides on the kubelet not the controller
+			// and is dynamically probed plugin, FindExpandablePluginBySpec will never find it.
+			// Therefore explicitly allow flex volumes to resize
+			volumePlugin, err := expc.volumePluginMgr.FindExpandablePluginBySpec(volumeSpec)
+			glog.V(4).Infof("plugin=%v err=%v fv=%v", volumePlugin != nil, err, volumeSpec.PersistentVolume.Spec.FlexVolume)
+			if err != nil || (volumePlugin == nil && volumeSpec.PersistentVolume.Spec.FlexVolume == nil) {
```

Comment:
`volumeSpec.PersistentVolume.Spec.FlexVolume == nil`   always be `true` at this line, should deleted it.

Reply:
sounds good.